azure-mgmt-website/src/main/java/com/microsoft/azure/management/website/Domain.java

```diff
@@ -0,0 +1,321 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.management.website;
+
+import com.microsoft.azure.management.resources.fluentcore.arm.models.GroupableResource;
+import com.microsoft.azure.management.resources.fluentcore.arm.models.HasName;
+import com.microsoft.azure.management.resources.fluentcore.model.Appliable;
+import com.microsoft.azure.management.resources.fluentcore.model.Creatable;
+import com.microsoft.azure.management.resources.fluentcore.model.Refreshable;
+import com.microsoft.azure.management.resources.fluentcore.model.Updatable;
+import com.microsoft.azure.management.resources.fluentcore.model.Wrapper;
+import com.microsoft.azure.management.website.implementation.DomainInner;
+import org.joda.time.DateTime;
+
+import java.util.List;
+
+/**
+ * An immutable client-side representation of a domain.
+ *
+ * Domains in Azure are purchased from 3rd party domain providers. By calling
+ * {@link Creatable#create()} or {@link Creatable#createAsync()} you agree to
+ * the agreements listed in {@link Domains#listAgreements(String)}.
+ */
+public interface Domain extends
+        GroupableResource,
+        HasName,
+        Refreshable<Domain>,
+        Updatable<Domain.Update>,
+        Wrapper<DomainInner> {
+
+    /**
+     * @return admin contact information
+     */
+    Contact adminContact();
+
+    /**
+     * @return billing contact information
+     */
+    Contact billingContact();
+
+    /**
+     * @return registrant contact information
+     */
+    Contact registrantContact();
+
+    /**
+     * @return technical contact information
+     */
+    Contact techContact();
+
+    /**
+     * @return domain registration status
+     */
+    DomainStatus registrationStatus();
+
+    /**
+     * @return name servers
+     */
+    List<String> nameServers();
+
+    /**
+     * @return true if domain privacy is enabled for this domain
+     */
+    boolean privacy();
+
+    /**
+     * @return domain creation timestamp.
+     */
+    DateTime createdTime();
+
+    /**
+     * @return domain expiration timestamp.
+     */
+    DateTime expirationTime();
+
+    /**
+     * @return timestamp when the domain was renewed last time
+     */
+    DateTime lastRenewedTime();
+
+    /**
+     * @return true if domain will renewed automatically
+     */
+    boolean autoRenew();
+
+    /**
+     * @return true if Azure can assign this domain to Web Apps. This value will
+     * be true if domain registration status is active and it is hosted on
+     * name servers Azure has programmatic access to.
+     */
+    boolean readyForDnsRecordManagement();
+
+    /**
+     * @return all hostnames derived from the domain and assigned to Azure resources
+     */
+    List<HostName> managedHostNames();
```

Comment:
how about Map<String, HostName> --- isn't there some useful unique string to index by?